### PR TITLE
Retire DEAL_II_COMPILE_EXAMPLES.

### PR DESCRIPTION
**Pull request by marcfehling**
Follow-up to #17486. 

Since we can now easily compile and run all examples in the testsuite, I suggest we retire the `DEAL_II_COMPILE_EXAMPLES` option, which was introduced in #3986.

I'm not sure what to do with the "Also compile some documentation examples" part. Do we build the documentation in `dealii/examples/doxygen` somewhere else?

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -46,105 +46,6 @@ if(DEAL_II_COMPONENT_EXAMPLES)
...
-          string(REPLACE "-" "_" _target "${_target}")
-          add_executable(${_target} ${_step})
-          insource_setup_target(${_target} ${_build})
-          set_target_properties(${_target}
-            PROPERTIES
-            RUNTIME_OUTPUT_NAME "${_name}.${_build_lowercase}"
-            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${DEAL_II_EXECUTABLE_RELDIR}"
-            )
-        endforeach()
-    endforeach()
```
Comment (marcfehling) [inline examples/CMakeLists.txt:145]: This is the part I was talking about.

**Review by tamiko — APPROVED** [review]
:musical_note:  Time to Say Goodbye :musical_note:

File: examples/CMakeLists.txt (hunk 1 above)
Comment (tamiko) [reply examples/CMakeLists.txt:145]: Let's not worry about this. We should convert these code snippets into full tests, or full example steps.

**Comment by tjhei** [comment]
@luca-heltai You need to be aware of this for Mac packaging.

**Comment by luca-heltai** [comment]
Perfect. I'll fix the spack package after they accept v9.6.0.

**Pull request merged**